### PR TITLE
fix: use `codersdk` functions for validating name attributes

### PR DESCRIPTION
**Pull request by aslilac**
rather than using regex that may drift over time, we should just use the functions provided by `codersdk` to validate name attributes

### CODE REVIEW

File: internal/provider/group_resource.go [hunk 1]
```diff
@@ -77,17 +77,15 @@ func (r *GroupResource) Schema(ctx context.Context, req resource.SchemaRequest,
 				MarkdownDescription: "The unique name of the group.",
 				Required:            true,
 				Validators: []validator.String{
-					stringvalidator.LengthBetween(1, 36),
-					stringvalidator.RegexMatches(nameValidRegex, "Group names must be alpahnumeric with hyphens."),
+					codersdkvalidator.Name(),
```
Comment (ethanndickson) [inline internal/provider/group_resource.go:80]: Group names need to use `codersdk.GroupNameValid` (which we recently bumped to like 255 characters)

File: internal/codersdkvalidator/validator_from_func.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	}
+}
+
+var _ validator.Describer = functionValidator{}
+
+func (v functionValidator) Description(_ context.Context) string {
+	if v.err != nil {
+		return v.err.Error()
+	}
+	return "value must be a valid name"
```
Comment (ethanndickson) [inline internal/codersdkvalidator/validator_from_func.go:46]: We should use the `defaultMessage` field here, right?

**Review by ethanndickson — APPROVED** [review]
Thanks for noticing this, this is much better! I actually moved the name validators from `coderd` into `codersdk` back in september and then forgot to use them here once the release came out :sob:

File: internal/codersdkvalidator/validator_from_func.go (hunk 2 above)
Comment (aslilac) [reply internal/codersdkvalidator/validator_from_func.go]: yes! good catch, I forgot the last step of this refactor lol

**Pull request merged**